shell/directives/clean-tooltip.ts

```diff
@@ -0,0 +1,271 @@
+import { DirectiveBinding, Directive } from 'vue';
+import { destroyTooltip, createTooltip } from 'floating-vue';
+import { purifyHTML } from '@shell/plugins/clean-html';
+
+// This is a singleton tooltip implementation that improves performance on pages with many tooltips.
+// Instead of instantiating a Vue component for every tooltip on the page, this directive attaches lightweight event listeners.
+// It then imperatively creates and destroys a single tooltip instance as needed, avoiding the high upfront memory and processing cost of many Vue components.
+let singleton: any = null;
```

Comment:
Instead of `any` we should be able to specify a type for `singleton` based off the return type of `createTooltip()`


```suggestion
let singleton: ReturnType<typeof createTooltip> | null = null;
```

Reply:
great! thanks for the code suggestion